File: repo-root/groups/group-01-manage-field/code/managefield/src/main/java/com/jad/efreifive/managefield/controller/ReservationController.java
package com.jad.efreifive.managefield.controller;

import com.jad.efreifive.managefield.dto.CreateReservationRequestDto;
import com.jad.efreifive.managefield.dto.ReservationDto;
import com.jad.efreifive.managefield.dto.UpdateReservationStatusRequestDto;
import com.jad.efreifive.managefield.mapper.ReservationMapper;
import com.jad.efreifive.managefield.service.ReservationService;
import com.jad.efreifive.managefield.vo.Id;
import com.jad.efreifive.managefield.vo.TimeSlot;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/v1/fields/{field_id}/reservations")
@RequiredArgsConstructor
public class ReservationController {

    private final ReservationService reservationService;
    private final ReservationMapper reservationMapper;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public ReservationDto createReservation(
        @PathVariable("field_id") String fieldId,
        @Valid @RequestBody CreateReservationRequestDto request
    ) {
        return reservationMapper.toDto(
            reservationService.createReservation(
                Id.of(fieldId),
                Id.of(request.getStatusId()),
                TimeSlot.of(request.getDate(), request.getStartTime(), request.getEndTime())
            )
        );
    }

    @PatchMapping("/{reservation_id}/status")
    public ReservationDto updateReservationStatus(
        @PathVariable("field_id") String fieldId,
        @PathVariable("reservation_id") String reservationId,
        @Valid @RequestBody UpdateReservationStatusRequestDto request
    ) {
        return reservationMapper.toDto(
            reservationService.updateReservationStatus(
                Id.of(fieldId),
                Id.of(reservationId),
                Id.of(request.getStatusId())
            )
        );
    }
}

File: repo-root/groups/group-01-manage-field/code/manage-field/src/main/kotlin/fr/efrei/managefield/controller/ReservationController.kt
package fr.efrei.managefield.controller

import fr.efrei.managefield.controller.dto.request.CreateReservationRequestDto
import fr.efrei.managefield.controller.dto.request.UpdateReservationStatusRequestDto
import fr.efrei.managefield.controller.dto.response.ReservationResponseDto
import fr.efrei.managefield.mapper.ReservationApiMapper
import fr.efrei.managefield.service.ReservationService
import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import org.springframework.http.ResponseEntity
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.PatchMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.net.URI

/**
 * Exposes HTTP endpoints for reservation creation and status changes.
 */
@Validated
@RestController
@RequestMapping("/v1/fields/{field_id}/reservations")
class ReservationController(
    private val reservationService: ReservationService,
    private val reservationApiMapper: ReservationApiMapper
) {
    @PostMapping
    fun create(
        @PathVariable("field_id") @NotBlank fieldId: String,
        @Valid @RequestBody request: CreateReservationRequestDto
    ): ResponseEntity<ReservationResponseDto> {
        val command = reservationApiMapper.toCreateCommand(fieldId, request)
        val response = reservationApiMapper.toReservationResponse(reservationService.create(command))

        return ResponseEntity
            .created(URI.create("/v1/fields/${response.fieldId}/reservations/${response.id}"))
            .body(response)
    }

    @PatchMapping("/{reservation_id}/status")
    fun changeStatus(
        @PathVariable("field_id") @NotBlank fieldId: String,
        @PathVariable("reservation_id") @NotBlank reservationId: String,
        @Valid @RequestBody request: UpdateReservationStatusRequestDto
    ): ReservationResponseDto {
        val command = reservationApiMapper.toChangeStatusCommand(fieldId, reservationId, request)
        return reservationApiMapper.toReservationResponse(reservationService.changeStatus(command))
    }
}
